openpype/hosts/aftereffects/plugins/publish/collect_workfile.py

```diff
@@ -49,14 +49,20 @@ def _get_new_instance(self, context, scene_file):
         asset_entity = context.data["assetEntity"]
         project_entity = context.data["projectEntity"]
 
+        handle_start = asset_entity["data"].get("handleStart") or context.data.get("handleStart")  # noqa
+        handle_end = asset_entity["data"].get("handeEnd") or context.data.get("handleEnd")  # noqa
+
+        if handle_start is None or handle_end is None:
+            raise KeyError("Handles are not set for this context")
+
         instance_data = {
             "active": True,
             "asset": asset_entity["name"],
             "task": task,
             "frameStart": asset_entity["data"]["frameStart"],
             "frameEnd": asset_entity["data"]["frameEnd"],
-            "handleStart": asset_entity["data"]["handleStart"],
-            "handleEnd": asset_entity["data"]["handleEnd"],
+            "handleStart": handle_start,
+            "handleEnd": handle_end,
```

Comment:
I think we could use only `context.data` where these keys should be validated and cleaned up.

```suggestion         
             "frameStart": context.data["frameStart"],
            "frameEnd": context.data["frameEnd"],
            "handleStart": context.data["handleStart"],
            "handleEnd": context.data["handleEnd"],
```